pkg/trace/api/version.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+// Version is a dumb way to version our collector handlers
+type Version string
+
+const (
+	// v01 DEPRECATED, FIXME[1.x]
+	// Traces: JSON, slice of spans
+	// Services: deprecated
+	v01 Version = "v0.1"
+
+	// v02 DEPRECATED, FIXME[1.x]
+	// Traces: JSON, slice of traces
+	// Services: deprecated
+	v02 Version = "v0.2"
+
+	// v03
+	// Traces: msgpack/JSON (Content-Type) slice of traces
+	// Services: deprecated
+	v03 Version = "v0.3"
+
+	// v04
+	// Traces: msgpack/JSON (Content-Type) slice of traces + returns service sampling ratios
+	// Services: deprecated
+	v04 Version = "v0.4"
+
+	// v05
+	//
+	// Content-Type: application/msgpack
+	// Payload: Traces with strings de-duplicated into a dictionary.
+	// Response: Service sampling rates.
+	//
+	// The payload is an array containing exactly 2 elements:
+	//
+	// 	1. An array of all unique strings present in the payload (a dictionary referred to by index).
+	// 	2. An array of traces, where each trace is an array of spans. A span is encoded as an array having
+	// 	   exactly 12 elements, representing all span properties, in this exact order:
+	//
+	// 		 0: Service   (int)
```

Comment:
Nit: What kind of `int` is that? Is that up to the sender (the tracer) to use the most efficient (compact) type of int?
If so, using `uint` could save a bit of space, on top of preventing crashes from `dict[i]` if the sent data is incorrect.

Reply:
Switched to `uint32`